cpp/src/skyhook/protocol/rados_protocol.cc

```diff
@@ -0,0 +1,112 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+#include "skyhook/protocol/rados_protocol.h"
+
+#include <iostream>
+#include <vector>
+
+namespace skyhook {
+namespace rados {
+
+RadosStatus GetStatusFromReturnCode(int code, std::string msg) {
+  if (code) return RadosStatus(arrow::Status::Invalid(msg), code);
+  return RadosStatus(arrow::Status::OK(), code);
+}
+
+RadosStatus IoCtxWrapper::write_full(const std::string& oid, ceph::bufferlist& bl) {
+  return GetStatusFromReturnCode(this->ioCtx->write_full(oid, bl),
```

Comment:
Nit: It's a little inconsistent that you start using the optional `this->` but only in this file.